packages/gcf-utils/test/gcf-logger.ts

```diff
@@ -0,0 +1,81 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {GCFLogger} from '../src/gcf-utils';
+import {describe, beforeEach, it} from 'mocha';
+import assert from 'assert';
+import {ObjectWritableMock} from 'stream-mock';
+import pino from 'pino';
+import {validateLogs, LogLine} from './test-helpers';
+
+describe('GCFLogger', () => {
+  describe('get()', () => {
+    it('returns a new pino-based logger instance on first call', () => {
+      const logger = GCFLogger.get();
+      assert.equal(logger.constructor.name, 'Pino');
+    });
```

Comment:
Personally, I think we should test behavior of the logger and not the class of the implementation.

Reply:
Well `GCFLogger.get()` has a return type of `pino.Logger` and so the caller expects everything that comes with a pino logger. So if we were to swap this out in the future for say Bunyan, and there were some Pino features missing in Bunyan, this test would tell us it's a breaking change. 

I could on the other hand test all the Pino features, but that would just be duplicating Pino's own tests and we'd have to add more every time they add more features.

Lastly, I did try to check `logger instanceof pino.Logger` but it wouldn't let me do that because `pino.Logger` is a `type` rather than a `class` or `interface`

Reply:
If you feel like we don't need to check the type for the logger, then I can remove this test. In fact, in that case, I might change it so that GCFLogger extends pino.Logger and then we completely abstract away pino from the bots

Reply:
With interfaces, the goal is that it shouldn't matter what the underlying implementation class is called. In some rare cases, you may want to know what the implementation is, but in this case, it really seems like we only care about the behavior.

If you do change the underlying implementation, tests usually shouldn't need to be rewritten to match the interface (sometimes mocks need to change though). Refactoring to extend pino.Logger, assuming it provides the same interface, shouldn't require touching the tests.

Reply:
Got it! My initial goal was to abstract away Pino completely but it would mean adding wrappers to all of Pino's methods. But I see that @bcoe has suggested exactly that below.

I'll incorporate both your comments here and push another commit to fix this. Thanks both!